### ISSUE TITLE
does deepstream eliminate the need for a real-time database and/or websockets???

### ISSUE THREAD

**Issue by quadsurf** [deepstreamIO/deepstream.io#526]
I'm new to coding, so please be detailed in your responses...

I'm very intrigued by this technology, and want to use it my next project, but first...

in a typical app, there are the following pieces:
- the DB server
- the web server app
- the client app

how many of the 3 pieces above need to be real-time in order for my app to truly be 100% real-time (ignoring network latency obviously)?

do all 3 pieces need to be real-time? for example:
DB server: rethinkDB
web server: deepstreamIO
client app: socksJS or socket.io or websockets

this seems rather redundant for all 3 pieces to be real-time

can deepstreamIO make my app fully real-time without having to use rethinkDB? if using mongo instead of rethink, then i'm losing real-time on the DB side, but does deepstreamIO make up for that? and if yes, how?

does a deepsteamIO with rethinkDB stack make an app more inherently real-time than a deepstreamIO with mongo stack? and if yes, how so?

and what about real-time on the client app side of things? can deepstreamIO eliminate the need for websockets in my client app, so that all i need are javascript listeners/subscribers? (i'm assuming web sockets is still needed, but its abstracted away by deepstreamIO, correct???)

in other words, could i achieve full real-time capabilities with deepstreamIO *alone*, like so:
DB server: mongoDB (not real-time)
web server: deepstreamIO
client app: listeners

**Comment by quadsurf** [comment]
answered by @WolframHempel 

- deepstream consists of a server and a client, so there's no need for socket.io etc.
- deepstream is not an HTTP server, so you'll need to serve your static assets (html, css, images etc) from something else
- deepstream can be used as a realtime layer in front of any database. As long as clients and backend processes interact with the db trough deepstream, all realtime functionality stays intact
- RethinkDB gives you queries with updating result sets, MongoDB doesn't. The rest is pretty much the same

please find more details at

https://deepstream.io/tutorials/guides/what-is-deepstream/
